### PR TITLE
MRG Fix corrmap no-plot bug

### PR DESCRIPTION
**Pull request by jona-sassenhagen**
Fix bug where corrmap with `plot=False, show=False` fails.

### CODE REVIEW

**Comment by jaeilepp** [comment]
I can't quite get my head around this. What does this function do if ``plot=False``?
Also, it seems the docs are messed up. At least cmap, sensors and layout are documented twice. Weird that tests didn't catch this.

**Comment by jona-sassenhagen** [comment]
The main purpose of the function is to change the supplied IC objects. The plot is just for diagnostics.

Standard operating procedure would be to run it a few times with labelling off and plots on to find good parameters, and then run it once (or more, if you want to detect multiple component types) with plot off and labelling on to actually change the IC objects.

**Comment by jaeilepp** [comment]
Okay, so I also need to set the ``label`` if I want something to happen. Should we maybe throw a warning if ``plot=False`` and ``label=None``? It doesn't do anything then, does it?

**Comment by jona-sassenhagen** [comment]
It's still computing correlations, which we could I guess print. (You could have a "headless" diagnostic mode, where only the subj/IC indices are printed.)
But as is, it's indeed only wasting CPU cycles if plot and label are both off.

**Comment by jaeilepp** [comment]
Yeah, since I'm not aware of the specific work flow of this, I'm not sure if this use case is something people will run to. We could also just update the docs or leave as it is. Your call.

**Comment by jona-sassenhagen** [comment]
I'd leave it as-is then and keep a mental note to add text print-out of ICs eventually, so plot + label off would be useful.

I've amended the docs a bit for an example of workflow.

**Comment by jaeilepp** [comment]
Thanks @jona-sassenhagen

File: mne/preprocessing/ica.py [hunk 1]
```diff
@@ -2349,16 +2354,6 @@ def corrmap(icas, template, threshold="auto", label=None,
...
-    layout : None | Layout | list of Layout
-        Layout instance specifying sensor positions (does not need to be
-        specified for Neuromag data). Or a list of Layout if projections
-        are from different sensor types.
-    cmap : matplotlib colormap
-        Colormap.
-    sensors : bool | str
-        Add markers for sensor locations to the plot. Accepts matplotlib plot
-        format string (e.g., 'r+' for red plusses). If True, a circle will be
-        used (via .add_artist). Defaults to True.
```
Comment (agramfort) [inline mne/preprocessing/ica.py]: the order of the documented parameters do not match the order on the function signature AFAIK

**Comment by agramfort** [comment]
thansk @jona-sassenhagen !

**Pull request merged**